ci/test-elk-flow-collector.sh

```diff
@@ -29,40 +76,35 @@ config_antrea() {
   sed -i -e "s/#flowCollectorAddr.*/flowCollectorAddr: \"${LOGSTASH_IP}:${LOGSTASH_PORT}:${LOGSTASH_PROTOCOL}\"/g" ${GIT_CHECKOUT_DIR}/build/yamls/antrea.yml
   sed -i -e "s/#  FlowExporter: false/  FlowExporter: true/g" ${GIT_CHECKOUT_DIR}/build/yamls/antrea.yml
   sed -i -e "s/#  AntreaProxy: false/  AntreaProxy: true/g" ${GIT_CHECKOUT_DIR}/build/yamls/antrea.yml
-  echo "=== Restarting Antrea ==="
-  kubectl apply -f ${GIT_CHECKOUT_DIR}/build/yamls/antrea.yml
-  sleep 20
-  kubectl wait --for=condition=ready pod -l app=antrea -n kube-system --timeout=300s
-  echo "=== Antrea Agent flow exporter starts to send data ==="
 }
 
 # Antrea agent flow exporter starts to send CoreDNS flow records.
 # It will check if flow records with one of desired fields (soursePodName) are received correctly.
 check_record() {
-  echo "=== Wait for 5 minutes to receive data ==="
-  sleep 5m
-   # if the records are received in logstash and processed correctly, the logstash logs should show the formatted data, which have 'sourcePodName' field
-  LOGSTASH_LOGS=$(kubectl logs -n antrea-flow-collector $(kubectl -n antrea-flow-collector get pod -l app=logstash -o jsonpath="{.items[0].metadata.name}"))
-  if ( echo ${LOGSTASH_LOGS} | grep -q 'sourcePodName' ); then
-    echo "=== Record is received correctly ==="
-  else
-    echo "=== Record is NOT received correctly ==="
-    FAILURE=true
-  fi
-}
-
-cleanup() {
-  echo "=== Removing Elastic Stack ==="
-  kubectl delete -f ${GIT_CHECKOUT_DIR}/build/yamls/flow-collector/flow-collector.yml  -n antrea-flow-collector
-  kubectl delete configmap logstash-configmap -n antrea-flow-collector
-  kubectl delete namespace antrea-flow-collector
-  echo "=== Elastic Stack has been removed ==="
+  echo "=== Wait for up to 5 minutes to receive data ==="
+  for i in `seq 5`
+  do
+    sleep 1m
+    echo "=== Get flow record (try for 1m) ==="
+    # if the records are received in logstash and processed correctly, the logstash logs should show the formatted data, which have 'sourcePodName' field
+    LOGSTASH_LOGS=$(kubectl logs -n elk-flow-collector $(kubectl -n elk-flow-collector get pod -l app=logstash -o jsonpath="{.items[0].metadata.name}"))
+    if ( echo ${LOGSTASH_LOGS} | grep -q 'sourcePodName' ); then
+      echo "=== Record is received correctly ==="
+      break
+    elif [ $i == 5 ]; then
+      echo "=== Record is NOT received correctly ==="
+      FAILURE=true
+    fi
+  done
 }
 
+start_antrea
 setup_flow_collector
 config_antrea
+start_antrea
 check_record
-cleanup
+
 if ( ${FAILURE} == true ); then
-  exit 1
+  echo "=== TEST FAILURE !! ==="
+  touch TEST_FAILURE
```

Comment:
Not sure why we need this line here

Reply:
That is suggested by @lzhecheng that we don't want to exit right away when it fails, but rather finish other jobs first.

Reply:
I meant `touch TEST_FAILURE`

Reply:
yes that's what I was referring to. I think that will work as indicator for failure status without exiting. @lzhecheng may provide more elaboration on that.

Reply:
I'm confused why we don't `exit 1` here. This script is used in `builder-elk-flow-collector` as `./ci/test-elk-flow-collector.sh --kubeconfig jenkins/out/kubeconfig`, so w/o exiting here this particular Jenkins job will never report fail status but only print TEST FAILURE?

Reply:
Hello @Dyanngg , this is a fix PR so you may not see its `builders` field in the elk Jenkins job. There is a cleanup builder after this builder-elk. If it exits 1 here, this build will stop without cleaning up the cluster on VMC.
https://github.com/vmware-tanzu/antrea/blob/master/ci/jenkins/jobs/projects.yaml#L49

Reply:
I see. Thanks for the explanation.